src/Subscriber/PaymentSubscriber.php

```diff
@@ -180,6 +182,10 @@ public function __construct(
         $this->cartCalculator = $cartCalculator;
         $this->currency = $currency;
         $this->logger = $logger;
+        $this->adyenPluginId = $this->pluginIdProvider->getPluginIdByBaseClass(
```

Comment:
@peterojo are you sure this should be done in the constructor because due to dependency injection etc. this might happen without it being needed. Maybe add a getAdyenPluginId that retrieves the plugin ID the first time and stores it as variable in the class once it has been resolved?

Reply:
@AndreasA for now there are only 2 public functions triggered by separate events and they are will both need to instantiate this property when they are called, but I think your suggestion is very good and we can add it as a future improvement if we continue to expand this class

Reply:
@peterojo Sure. Maybe it would actually make sense to create a computed/runtime property on the payment method itself that provides adyen information like: if it is an adyen payment method and/or what type. Could be useful upon retrieving the payment method using the store api too.

I mostly mentioned it because Symfony does a lot of stuff upon determining its dependency injection build process, e.g. service inlining if a service is only used once. Shouldn't be an issue but just to bear in mind 😄